common/transport/mqtt/src/mqtt_base.ts

```diff
@@ -75,20 +118,31 @@ export class MqttBase extends EventEmitter {
           '*': () => this._fsm.deferUntilTransition()
         },
         connected: {
-          _onEnter: (connectCallback, conack) => {
-            this._mqttClient.on('close', () => {
-              debug('close event received from mqtt.js client - no error');
-              this._fsm.handle('closeEvent');
-            });
-            connectCallback(null, new results.Connected(conack));
+          _onEnter: (connectCallback, conAck) => {
+            this._mqttClient.on('close', this._closeCallback.bind(this));
+            connectCallback(null, new results.Connected(conAck));
           },
           connect: (callback) => callback(null, new results.Connected()),
           disconnect: (callback) => this._fsm.transition('disconnecting', callback),
           publish: (topic, payload, options, callback) => {
+            const thisPublishIdentifier = uuid.v4();
+            this._publishesOnTheWire[thisPublishIdentifier] = new OnTheWirePublish(Math.floor( Date.now() / 1000 ), topic, payload, options, callback, thisPublishIdentifier);
```

Comment:
it seems like topic, payload, options, thisPublishIdentifier are unnecessary to add here?

Reply:
You're right.  Didn't know when I started.  I can cut this and will.

---
In reply to: [431361482](https://github.com/Azure/azure-iot-sdk-node/pull/808#discussion_r431361482) [](ancestors = 431361482)